src/cart/auth/auth.controller.ts

```diff
@@ -0,0 +1,41 @@
+import express, { Response, Request } from "express";
+import { authService } from "./auth.service";
+import { checkLoginReq, checkRegisterReq } from "../../middlewares/auth.validator";
+
+const router = express.Router();
+
+router.post('/login', checkRegisterReq, async (req: Request, res: Response) => {
+    try {
+        const token = await authService.login(req.body);
+
+        if (token) {
+            res.json({data: token, error: null});
+            return;
+        }
+        res.status(401);
+        res.send({data: null, error: 'User is not authorizes'});
+    } catch (error) {
+        res.status(500);
+        res.send({data: null, error: 'something went wrong'});
+    }
+
+});
+
+router.post('/register', checkLoginReq, async (req: Request, res: Response) => {
+    try {
+        const user = await authService.register(req.body);
+
+        if (user) {
+            res.json({data: user, error: null});
+            return;
+        }
+        res.status(401);
```

Comment:
I think the case `!user` means that such user lready exists, so we should correct a bit error message and status code for response

![image](https://github.com/314olas/nodejs-run/assets/150139739/52d1f0ac-4161-4901-9541-4ea52a003c21)